### DISCUSSION TITLE
Fixers don't care about indentation

### DISCUSSION THREAD

**Discussion by lkster** [eslint/eslint#16062] [Q&A]
I was trying to create fixer for my custom rule in company's project but it seems I can't consistently indent added brace - there's just no info about trivia (i.e. tabs, spaces and other junk before the statement). So then I tried to look up how ESLint handles indentation by taking `object-curly-newline` rule's fixer as an example.  But this is how object is being fixed:

<table style="width: 100%">
<tbody>
<tr><td>
<img width="441" height="1">

```ts
function test() {
    const a = {
        a: 'a',
        b: 'b'}
}
```
</td>
<td>
<img width="441" height="1">

```ts
function test() {
    const a = {
        a: 'a',
        b: 'b'
}
}
```
</td></tr>
</tbody>
</table>

Or even better:

<table style="width: 100%">
<tbody>
<tr><td>
<img width="441" height="1">

```ts
function test() {
    const a = {a: 'a',
        b: 'b'}
}
```
</td>
<td>
<img width="441" height="1">

```ts
function test() {
    const a = {
a: 'a',
        b: 'b'
}
}
```
</td></tr>
</tbody>
</table>

Am I doing somethig wrong or ESLint really doesn't care about the indentation during fixing stuff?

EDIT: Ok I see now `indent` rule fixes indentations. But still shouldn't every rule care about such basic indentation when fixing it's stuff?

**Comment by nzakas** [comment]
No. Every rule can’t care about indentation because those preferences aren’t accessible. Just fix the syntax the way you need it and the indent rule will also run and fix any further issues afterwards.

**Reply by lkster** [reply]
Well as it's not a problem for me I found out that fixing with IDE's sugestions feature (WebStorm in my case) fixes in a way I showed i.e. doesn't apply indent rule as next step after using the other. What should be done in such situation? Should ESLint allow to mark that eg. after using fixer of some rule, other rule's fixer also needs to be applied (indent in this case)? Or WebStorm and other IDEs should handle this on their own with applying indent rule's fixer?
